definitions/npm/@reduxjs/toolkit_v1.x.x/flow_v0.104.x-/toolkit_v1.x.x.js

```diff
@@ -0,0 +1,194 @@
+     * @param matcher A matcher function. In TypeScript, this should be a [type predicate](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates)
+     *   function
+     * @param reducer
+     */
+    // addMatcher<A extends AnyAction>(matcher: ActionMatcher<A>, reducer: CaseReducer<State, A>): Omit<ActionReducerMapBuilder<State>, 'addCase'>;
+    /**
+     * Adds a "default case" reducer that is executed if no case reducer and no matcher
+     * reducer was executed for this action.
+     * @param reducer
+     */
+    // addDefaultCase(reducer: CaseReducer<State, AnyAction>): {};
+  }
+
+  /**
+   * An action with a string type and an associated payload. This is the
+   * type of action returned by `createAction()` action creators.
+   *
+   * @template P The type of the action's payload.
+   * @template T the type used for the action type.
+   *
+   * @public
+   */
+  declare type PayloadAction<P = void, T = string> = {|
+    payload: P,
+    type: T,
+  |};
+
+  /**
+   * An *case reducer* is a reducer function for a specific action type. Case
+   * reducers can be composed to full reducers using `createReducer()`.
+   *
+   * Unlike a normal Redux reducer, a case reducer is never called with an
+   * `undefined` state to determine the initial state. Instead, the initial
+   * state is explicitly specified as an argument to `createReducer()`.
+   *
+   * In addition, a case reducer can choose to mutate the passed-in `state`
+   * value directly instead of returning a new state. This does not actually
+   * cause the store state to be mutated directly; instead, thanks to
+   * [immer](https://github.com/mweststrate/immer), the mutations are
+   * translated to copy operations that result in a new state.
+   *
+   * @public
+   */
+  declare type CaseReducer<S = any, A = Action<string>> = (state: S, action: A) => S | void;
+
+  /**
+   * A mapping from action types to case reducers for `createReducer()`.
+   *
+   * @deprecated This should not be used manually - it is only used
+   *             for internal inference purposes and using it manually
+   *             would lead to type erasure.
+   *             It might be removed in the future.
+   * @public
+   */
+  declare type CaseReducers<S, AS = Actions<string>> = $ObjMapi<AS, (K, V) => {| [K]: CaseReducer<S, V> |}>;
+
+  /**
+   * Defines a mapping from action types to corresponding action object shapes.
+   *
+   * @deprecated This should not be used manually - it is only used for internal
+   *             inference purposes and should not have any further value.
+   *             It might be removed in the future.
+   * @public
+   */
+  declare type Actions<T = string> = Record<T, Action>;
+
+
+  /**
+   * Basic type for all action creators.
+   *
+   * @inheritdoc {redux#ActionCreator}
+   */
+  declare interface BaseActionCreator<P, T = string> {
+    type: T;
+    match: Action<unknown> => boolean;
```

Comment:
AFAIK there's no `unknown` type in Flow. Use `mixed` instead?

Reply:
Left over from TS defs. Why didn't Flow say it doesn't exist...? I'll change it to `mixed`.